core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -607,6 +616,15 @@ public DeterministicKey currentKey(KeyChain.KeyPurpose purpose) {
         }
     }
 
+    public DeterministicKey currentCoinJoinKey() {
+        keyChainGroupLock.lock();
+        try {
+            return coinJoinKeyChainGroup.currentKey(KeyChain.KeyPurpose.RECEIVE_FUNDS);
+        } finally {
+            keyChainGroupLock.unlock();
+        }
+    }
+
```

Comment:
This is an example of many functions for coinjoin that are copy/pasted from "regular" functions.  There may be 3 different `currentKey` functions now.  

My idea to remove this duplication is to create a few more classes that will encapsulate other keychains for coinjoin and dashpay related functions.